src/custom/state/ethFlow/updater.tsx

```diff
@@ -0,0 +1,20 @@
+import { Percent } from '@uniswap/sdk-core'
+import { useEffect } from 'react'
+import { useSetUserSlippageTolerance } from '../user/hooks'
+import { useIsUserNativeEthFlow } from './hooks'
+
+export const ETH_FLOW_SLIPPAGE = new Percent(2, 100) // 2%
+
+export default function Updater() {
+  const isUserNativeEthFlow = useIsUserNativeEthFlow()
+  const setUserSlippageTolerance = useSetUserSlippageTolerance()
+  // change slippage set to 2% when detected eth swap and option is set to use native flow
+  useEffect(() => {
+    if (isUserNativeEthFlow) {
+      setUserSlippageTolerance(ETH_FLOW_SLIPPAGE)
```

Comment:
Hm... It seems like this would set the slippage tolerance also for all erc20 trades to 2%. Probably, we need to make this logic dependent on whether the user has eth selected in the swap module. Or just handle everything in the swap module and not make any changes TransactionsSetting module... Not sure.

If you wanna discuss, ping me.

Reply:
good catch! fixing...

Reply:
fixed: [c0d1b91](https://github.com/cowprotocol/cowswap/pull/902/commits/c0d1b915c573dd8a4af801d16d95fbd47107a520)